code/modules/mob/living/living.dm

```diff
@@ -29,6 +29,13 @@
 	med_hud_set_status()
 
 /mob/living/Destroy()
+	if(LAZYLEN(status_effects))
+		for(var/s in status_effects)
+			var/datum/status_effect/S = s
+			if(S.on_remove_on_mob_delete) //the status effect calls on_remove when its mob is deleted
+				qdel(S)
+			else
+				S.be_replaced()
```

Comment:
It's just i don't feel the name of the function represents the use well if it's now the default removal proc.